src/neuroconv/tools/testing/data_interface_mixins.py

```diff
@@ -128,29 +176,64 @@ def check_align_starting_time_internal(self):
         expected_timestamps = unaligned_timestamps + starting_time
         assert_array_equal(x=aligned_timestamps, y=expected_timestamps)
 
-    def check_align_timestamps_internal(self):
-        """Ensure that internal mechanisms for the timestamps getter/setter work as expected."""
-        unaligned_timestamps = self.interface.get_original_timestamps()
+    def check_interface_align_timestamps_mutation(self):
```

Comment:
`check_interface_get_original_timestamps_inumability` seems better.

I see why you want to have this test (I guess you want to force people who implement this for a new interface in the future to recreate)  but I still find this strange to talk about the mutability of a the value returned by a function.

Maybe this hint that this should be a property that is read-only so this contract is enshrined in code?

Reply:
It is indeed supposed to be read-only, but how can you control that in a programmatic way for a newly created interface?

Reply:
Yeah, I know how to do it. But I agree that it is better to leave it for another PR.